src/inmanta/data/__init__.py

```diff
@@ -87,7 +87,7 @@
 are as follows. This list should be extended when new locks (explicit or implicit) are introduced. The rules below are written
 as `A -> B`, meaning A should be locked before B in any transaction that acquires a lock on both.
 - Code -> ConfigurationModel
-- Agent -> Agentprocess -> Agentinstance
+- Agentprocess -> Agentinstance -> Agent
```

Comment:
This affects `Environment.clear`  , `Agent.update_primary`  (possibly others, these are the ones I know/could find quickly)